### ISSUE TITLE
Search, sort and filter layout - update

### ISSUE THREAD

**Issue by robertocarroll** [robertocarroll/icc-alpha#60]
Some updates to the original layout proposed for the search, sort and filter page. 

![search-d1-r3](https://f.cloud.github.com/assets/1565953/1814351/45b6b4bc-6ee9-11e3-8931-fe00ede6a0ed.jpg)

I've added elements to the left column (which emerged from the way TRIM operates at the moment) - see issue #59 

I've also included other details in the results. Please let me know what you think.

**Comment by robertocarroll** [comment]
I've updated the layout to respond to a few issues: 
- situation and case - two separate fields 
- document number - 1 field 
- remove victim code 
- appeal is a text field 

Also we need to consider how it will work when a user comes from a pre-filtered link, eg clicking on see all filings on the Lubanga case page. 

Can the user then change the case and situation or will those fields not be available?

**Comment by simmigaba** [comment]
In my viewpoint, when user comes from a per-filtered link, we will populate those filter values as default selected in drop down e.g. appropriate Case and Situation value in the example you mentioned. User then can select other drop down/ enter text field values to refine the search results further. But should also be able to change Case, Situation value if user wish to do so. Thoughts?

**Comment by robertocarroll** [comment]
I agree. The user should be able to change the situation and case even if they come through on a filtered link. 

Just stepping back a bit and taking a broader look at this: If we go back to the sitemap, there's a handful of root pages which mark the beginning of a user's filter journey.

The main one that comes to mind is court records. The user can see all court records (any situation, any case) and then filter by various metadata on the left. 

Another is press releases. (There's still some debate about whether press releases should be split into judicial and non-judicial, but I feel they should have a different identity, eg press releases and ICC updates or something like that.) This would be another root page.

Another is publications, ie everything which is published by the ICC which is not a court record.   

Each of these would be the starting point for a filter journey, but there could be links to specific filtered parts from various places on the site, eg see all link below Lubanga defence would go to the court records template filtered by DRC, Lubanga, defence. 

I hope this hasn't confused everything. I was just trying to get it straight in my head and found this way of thinking about it useful.

**Comment by simmigaba** [comment]
Hi Rob - The All Filings page is WIP and can be viewed at: http://sharepoint13vm.cloudapp.net/sites/icc/allfilings
Functionality implemented:
-- Display default list of all Filings sorted by Sequence Number
-- Populate all the filter values (for drop downs) in Left Panel and dynamically filter the Filings results in Right Panel (Ajax) based on one or more Filter values (dropdown, text box) selected

To do:
-- Pagination logic (10 filings per page)
-- Search terms logic implementation
-- Linking from other pages in sites

For 'Search Terms', I need clarity regarding what all kind of searches user can do using this. Can you please elaborate. Thanks

**Comment by robertocarroll** [comment]
Just consider the search terms and there's a bit of an issue: 

Document number and Appeal number are free text boxes (as is the search terms). Putting a document number in the search terms box seems just as valid as putting any other kind of term ... Do we need to get rid of 
Document number and Appeal number and include them in the general search terms box? 

I will ask about some example search terms which can be used for testing.

**Comment by robertocarroll** [comment]
I have some example keyword searches for the Lubanga case: 
- 1. Keyword search on the following terms:  

Rwampara camp, bodyguard, training camp, co-perpetrators, mental element 

We could run these searches and cf Trim internal results to see how the search is working.
- 2. The following are witness numbers which a user might enter in the keyword search on Lubanga:

P-0014, P-0030, P-0046, EVD-OTP-00570 
- 3. Document number: Enter a document number and that document would appear first followed by a list of all documents that could refer to that document. This would eliminate the need for an extra field.

**Comment by simmigaba** [comment]
Thanks for providing further information.

**Comment by robertocarroll** [comment]
Here's some user feedback on the search / filter template:
1. Filter titles are way too big. Need to be able to see as many filters as possible onscreen at once. (Suggest 14px?)
2. Paginatiion needs to be more per page - more like 30 or 50 on each page.
3. Can we sort the results? Date / relevance 

Let's discuss and then decide what can / should be changed.

**Comment by simmigaba** [comment]
Hi Rob, The changes suggested by you looks reasonable hence I have implemented the same. Changes done on http://sharepoint13vm.cloudapp.net/sites/icc/allfilings page:
1. Filter titles font and spacing is modified for better viewing
2. Pagination is changed to 30 per page. 
3. The filing results are now sorted as per Publishing date (descending order). The order is maintained in filtered results. 
Pls feel free to suggest any further improvements based on user feedback. Thanks.

**Comment by robertocarroll** [comment]
Great! Thanks so much for this. A few more elements to consider to raise this layout to a really good standard: 
1. Can we can it "Court Records" - so the title at the top says court records, the breadcrumb says court records, and the url says /court-records. 
2. The line length in the results display is too long making it difficult to parse. Can we use Kyle's grid from the case page to create filters on the left and results on the right, but with the results width never exceeding about 640px? 
3. Next / Previous needs to only appear when results are > 30. At the moment they don't do anything because the results are 18. 
4. Can we include the filter default rather than a blank, eg Case would say "All cases", Phase would say "All phases" - it would help to give the user a sense of what the current display is and that it can be filtered.

**Comment by robertocarroll** [comment]
As always, if any of the suggestions above is more than a quick tweak - hold off and we can discuss further. thanks.

**Comment by simmigaba** [comment]
Hi Rob,
Pls note the changes done:
1. Now All filings Page URL is: http://sharepoint13vm.cloudapp.net/sites/icc/court-records
Also Bread crumb, Left side Menu and Title have value as "Court Records"
2. I have adjusted the width but as a relative %. Though I am not a UI design expert but my viewpoint is that fixed widths in px is never a good idea. Many smartphones have max standard width as 320 px or 480 px. Hence in all the layouts that I designed so far I am only using relative width/ height in % so that it is device agnostic. Only exception is images which have width/ height as fixed px size but images can also be displayed using relative width/ height.
3. Next/ Previous buttons logic is changed to be visible only when filtered/total number of court records > 30
4. Filter default now have values like "All cases", "All phases", etc.

Pls review and let me know. Thanks.
